tests/commands/record/test_build.py

```diff
@@ -29,7 +29,20 @@ def test_submodule(self, mock_check_output):
             ('c50f5de0f06fe16afa4fd1dd615e4903e40b42a2 refs/head/main\nc50f5de0f06fe16afa4fd1dd615e4903e40b42a2 refs/remotes/origin/main\n').encode(),  # noqa: E501
         ]
 
-        result = self.cli("record", "build", "--no-commit-collection", "--build", self.build_name)
+        result = self.cli(
+            "record",
+            "build",
+            "--no-commit-collection",
+            "--build",
+            self.build_name,
+            "--branch",
+            "main",
+            "--repo-branch-map",
```

Comment:
Q: Will we require branch name when the user sets `--no-commit-collection`, right? (Just in case)

Reply:
Yes, that's right.